core/src/main/java/org/infinispan/functional/impl/MetaParamsExternalizer.java

```diff
@@ -0,0 +1,39 @@
+package org.infinispan.functional.impl;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+import org.infinispan.commons.api.functional.MetaParam;
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+
+public final class MetaParamsExternalizer extends AbstractExternalizer<MetaParams> {
```

Comment:
Rather than having a specific `MetaParamsExternalizer`, it'd be more useful to have a `MarshallUtil.marshallIterable`...etc that can work for any Iterable, not just MetaParams. The externalizer is not doing anything special you'd not do for any other Iterable.